### PR TITLE
FEATURE: Pull Request Syncing

### PR DESCRIPTION
**Pull request by danielwaterworth**
WIP

### CODE REVIEW

File: lib/discourse_code_review/github_category_syncer.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+module DiscourseCodeReview
+  module GithubCategorySyncer
+    GithubRepoName = "GitHub Repo Name"
+
+    class << self
+      def ensure_category(repo_name:)
+        Category.transaction(requires_new: true) do
+          category =
+            Category.where(
```
Comment (SamSaffron) [inline lib/discourse_code_review/github_category_syncer.rb]: Technically you can just do this in one query... no need to run 2. 

`.where('id in (select category_id from category_custom_fields ....)').first`

also... be sure to use `find_by` to save the .first at then end for field lookups.

**Comment by SamSaffron** [comment]
tests are tiny bit thin, nonetheless super duper keen to see it in action! very much excite.

File: lib/discourse_code_review/github_category_syncer.rb (hunk 1 above)
Comment (danielwaterworth) [reply lib/discourse_code_review/github_category_syncer.rb]: @SamSaffron, here's the clever bit, it's already just doing one query (and it looks very similar to your SQL).

**Comment by danielwaterworth** [comment]
Yep, more on the way. Just wanted make my work public.

File: lib/discourse_code_review/github_category_syncer.rb (hunk 1 above)
Comment (SamSaffron) [reply lib/discourse_code_review/github_category_syncer.rb]: I guess if `where` get a relation AREL is sprinkling on even more magic and weaving this stuff together, I am willing to bet this benches slower not that it matter, I guess it is OK

**Comment by eviltrout** [comment]
This is quite a large PR and I assume Sam did the bulk of the reviewing already. We should probably merge when you are comfortable with it and test/fix it avidly on review.discourse to find any issues.

**Comment by SamSaffron** [comment]
I feel we can not wait on this for too long, @danielwaterworth  maybe merge at the beginning of your day ... deploy on review and watch it like a hawk?

**Comment by danielwaterworth** [comment]
Merged

**Pull request closed without merging**